### ISSUE TITLE
undefined method `visit_type' exception in messages compiler 

### ISSUE THREAD

**Issue by Kukunin** [dry-rb/dry-validation#260]
First of all, thank you very much for new updates.

Just tried `dry-validation` 0.10.1 version, and got exception from my schemas, when use `#messages` method on failed schema. 

Exception is this:

```
    NoMethodError:
       undefined method `visit_type' for #<Dry::Validation::MessageCompiler:0x007fa32373fe38>
       Did you mean?  visit_key
                      visit_set
                      visit
                      visit_not
                      visit_rule
     # /Users/kukunin/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.10.1/lib/dry/validation/message_compiler.rb:43:in `visit'
     # /Users/kukunin/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.10.1/lib/dry/validation/message_compiler.rb:137:in `visit_key'
     # /Users/kukunin/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.10.1/lib/dry/validation/message_compiler.rb:43:in `visit'
     # /Users/kukunin/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.10.1/lib/dry/validation/message_compiler.rb:73:in `visit_rule'
     # /Users/kukunin/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.10.1/lib/dry/validation/message_compiler.rb:43:in `visit'
     # /Users/kukunin/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-validation-0.10.1/lib/dry/validation/message_compiler.rb:81:in `block in visit_and'
```

Simple code to reproduce:

```ruby
        Types = Dry::Types.module
        MySchema = Dry::Validation.Schema do
          required(:address).schema do
            required(:zip).filled(Types::Strict::String.constrained(format: /\A[0-9]{5}(-[0-9]{4})?\z/))
          end
        end

        expect(MySchema.call(address: {zip: '91403'})).to be_success
        expect(MySchema.call(address: {zip: '1234'})).to be_failure
        expect(MySchema.call(address: {zip: '1234'}).messages).not_to be_empty #works fine
        expect { MySchema.call(address: nil).messages }.not_to raise_error #raises exception
        expect { MySchema.call({}).messages }.not_to raise_error #raises exception
```

It works fine when there is nested schema or I specify filled with predicate like `.filled(:str?)`

Hopefully these details are enough to locate and resolve this bug. 
Otherwise, release looks great. Thanks!

**Comment by solnic** [comment]
Integration with dry-types is still in experimental phase so we're missing some coverage (hence the regression). I'll deal with this in 0.10.2. Sorry about that.

**Comment by Kukunin** [comment]
@solnic is there a chance when http://dry-rb.org/gems/dry-validation/type-specs/ would help?

**Comment by solnic** [comment]
Not in this case. I'll fix it later today and push 0.10.2.

**Comment by solnic** [comment]
Seems like it's choking on hint compilation, to easily avoid this crash, please just use the new `errors` interface, so `MySchema.call({}).errors` should work. I'll fix hint compilation nevertheless of course.

**Comment by solnic** [comment]
Fixed in 0.10.2

**Comment by Kukunin** [comment]
Super! Thanks for quick fix!
